extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -1274,6 +1274,12 @@ export class CodeQLCliServer implements Disposable {
     suite: string,
     additionalPacks: string[],
   ): Promise<ResolveExtensionsResult> {
+    if (!(await this.cliConstraints.supportsResolveExtensions())) {
+      throw new Error(
+        "Resolving extensions is only supported for CodeQL CLI v2.10.2 or later",
+      );
+    }
+
```

Comment:
It seems like we never have a check in `cli.ts` for whether a method is supported which throws an error. We only seem to use it if certain arguments are not supported or if the arguments are different, but in those cases we only log or change the arguments. Could we remove this and instead rely on the consumer to do this check?

Reply:
I wasn't sure what to do here. I agree there aren't any other examples of a method throwing an error, and all the others fall back to another implementation if the version is too low. I wasn't sure if that was a deliberate design choice, or just coincidence that all other examples of versions checks have a way around them.

But I'm happy to remove this and rely on the caller to check compatibility. If it does fail then the error is pretty clear and we'd know what was wrong.